File: src/main/java/com/feiliks/blog/repositories/TagRepository.java
package com.feiliks.blog.repositories;

import java.util.Collection;

import com.feiliks.blog.entities.Tag;
import org.springframework.data.repository.CrudRepository;

public interface TagRepository extends CrudRepository<Tag, Long> {

    Tag findByName(String tag);

    boolean existsByName(String tag);

    Collection<Tag> findOrphaned();
}

File: src/main/kotlin/com/feiliks/blog/repositories/TagRepository.kt
package com.feiliks.blog.repositories

import com.feiliks.blog.entities.TagEntity
import org.springframework.data.repository.CrudRepository

interface TagRepository : CrudRepository<TagEntity, Long> {

    fun findByName(tag: String): TagEntity?

    fun existsByName(tag: String): Boolean

    fun findOrphand(): Set<TagEntity>

}
